docs/developer-guide/ksql-testing-tool.rst

```diff
@@ -200,3 +200,9 @@ the testing tool first processes the input messages provided in the input file.
 the testing tool inspects the source topics for the query in the simulated Kafka cluster and processes any messages in these topics.
 For the JOIN queries where we have more than one source topic, the testing tool first processes the left-side topic and
 then processes the right hand side topic.
+
+UDFs and UDAFs
+--------------
+
+The KSQL testing tool supports UDFs and UDAFs the same as KSQL. However, unlike KSQL where the location of the jar file that contains the UDF/UDAF code is configured by ``ksql.extension.dir``,
+in the KSQL testing tool, you should put the jar file in the same folder as the test statements file.
```

Comment:
this doesn't seem like a very good user experience. I should be able to point you to where the JAR is instead of forcing me to place it in a specific place for tests (likely duplicating where I actually store it) by providing a `--udf-jar path/to/jar` flag or something like that

Reply:
(replying to Almog's thread: https://github.com/confluentinc/ksql/pull/3252/files#r316900701)

Totally agree - we should not require the jar in the same folder as the statements file.

We should be able to run the test with a directory structure where test files are in sub-directories, and we shouldn't expect people to copy the jar into multiple places.

I would say we just load it from the class path.  There is no need to do anything special if we do that.  

I'm not sure we need a `--udf-jar path/to/jar` flag for the application, the same can be achieved by adding it to the class path on the command line.  Of course we could have a wrapper script, (we may already!), that took such a parameter and included it in the class path.

Side note: Can we currently run the tool against a directory of files or are we expecting users to script up reading a directory of files and invoking per-file?  Surely the former... otherwise just the cost of spinning up the JVM for each test is going to be prohibitive!